integration/spark2/src/main/scala/org/apache/spark/sql/CarbonEnv.scala

```diff
@@ -121,11 +125,30 @@ class CarbonEnv {
         CarbonProperties.getInstance
           .addNonSerializableProperty(CarbonCommonConstants.IS_DRIVER_INSTANCE, "true")
         initialized = true
+        cleanChildTablesNotRegisteredInHive(sparkSession)
       }
     }
     Profiler.initialize(sparkSession.sparkContext)
     LOGGER.info("Initialize CarbonEnv completed...")
   }
+
+  private def cleanChildTablesNotRegisteredInHive(sparkSession: SparkSession): Unit = {
+    val dataMapSchemas = DataMapStoreManager.getInstance().getAllDataMapSchemas
+    dataMapSchemas.asScala.foreach {
+      dataMapSchema =>
+        if (null != dataMapSchema.getRelationIdentifier &&
+            !dataMapSchema.isIndexDataMap) {
+          if (!sparkSession.sessionState
+            .catalog
+            .tableExists(TableIdentifier(dataMapSchema.getRelationIdentifier.getTableName,
+              Some(dataMapSchema.getRelationIdentifier.getDatabaseName)))) {
+            DataMapStoreManager.getInstance().dropDataMapSchema(dataMapSchema.getDataMapName)
+            DataMapStoreManager.getInstance.unRegisterDataMapCatalog(dataMapSchema)
+            FileUtils.deleteDirectory(new File(dataMapSchema.getRelationIdentifier.getTablePath))
```

Comment:
Instead of Directly using FileUtils, can't we use, `CarbonUtil.deleteFoldersAndFilesSilent` ?

Reply:
ok..done